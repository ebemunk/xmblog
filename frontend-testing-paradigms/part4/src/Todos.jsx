import React from "react";

import Item from "./Item";

const Todos = ({ list }) => (
  <div className="todos">
    {list.map(item => (
      <Item
        key={item.name}
        item={item}
        onChange={() => {
          // do things
        }}
      />
    ))}
  </div>
);

export default Todos;
