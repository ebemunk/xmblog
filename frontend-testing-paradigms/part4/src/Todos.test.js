import React from "react";
import { render } from "react-testing-library";

import Todos from "./Todos";

const items = [
  { name: "one", complete: false },
  { name: "two", complete: true },
  { name: "three", complete: true },
  { name: "four", complete: false }
];

test("renders the items - SNAPSHOT", () => {
  const { container } = render(<Todos list={items} />);

  expect(container.firstChild).toMatchSnapshot();
});

test("renders the items", () => {
  const { container, getByText } = render(<Todos list={items} />);

  expect(container.querySelectorAll(".item").length).toBe(4);
  expect(container.querySelectorAll(".complete").length).toBe(2);

  expect(container.querySelectorAll(".item")[1].className).toBe(
    "item complete"
  );
  expect(container.querySelectorAll(".item")[2].className).toBe(
    "item complete"
  );

  // We need to add even more tests now to avoid this kind of error.
  // Notice how the Snapshot test diff is much more descriptive
  // and less tedious to write and maintain
  items.forEach(item => {
    expect(getByText(item.name)).toBeDefined();
  });
});
