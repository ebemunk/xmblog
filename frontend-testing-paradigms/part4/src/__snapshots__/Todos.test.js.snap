// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the items - SNAPSHOT 1`] = `
<div
  class="todos"
>
  <li
    class="item"
  >
    <input
      type="checkbox"
    />
    one
  </li>
  <li
    class="item complete"
  >
    <input
      checked=""
      type="checkbox"
    />
    two
  </li>
  <li
    class="item complete"
  >
    <input
      checked=""
      type="checkbox"
    />
    three
  </li>
  <li
    class="item"
  >
    <input
      type="checkbox"
    />
    four
  </li>
</div>
`;
